src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -218,13 +224,8 @@ private Task InternalWriteAsync(byte[] buffer, int offset, int count, Cancellati
 
         private Task<bool> InternalWriteDataAsync(byte[] buffer, int offset, int count, CancellationToken token)
         {
-            Debug.Assert(count >= 0);
-            
-            if (count == 0)
-            {
-                return Task.FromResult<bool>(true);
-            }
```

Comment:
>any value in this fast path still being kept for that?

No.  EndUploadDataAsync is only called in chunked mode *and* it only calls it with a buffer that is always > 0 in length.

```c#
        internal async Task EndUploadAsync(CancellationToken token)
        {
            if (_chunkedMode)
            {
                await InternalWriteDataAsync(s_endChunk, 0, s_endChunk.Length, token).ConfigureAwait(false);
            }
        }
```